x-pack/metricbeat/module/awsfargate/task_stats/task_stats.go

```diff
@@ -104,8 +118,12 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 }
 
 func (m *MetricSet) queryTaskMetadataEndpoints() ([]Stats, error) {
-	// Get response from ${ECS_CONTAINER_METADATA_URI_V4}/task/stats
-	taskStatsResp, err := http.Get(m.taskStatsEndpoint)
+	// Collect information from ${ECS_CONTAINER_METADATA_URI_V4}/task/stats
```

Comment:
This code and the task metadata endpoint both run inside the same cluster, but since we're calling an external service, we should consider adding a context with a timeout.

WDYT?

Reply:
Agree :). Fixed in https://github.com/elastic/beats/pull/32342/commits/d1ed939cbd80ce455b154968758430d6693e4797